crates/python/README.md

```diff
@@ -7,3 +7,58 @@ to compile and run Quil programs on Rigetti quantum processors. Internally, it i
 
 While this package can be used directly, [pyQuil](https://pypi.org/project/pyquil/) offers more functionality and a 
 higher-level interface for building and executing Quil programs. This package is still in early development and breaking changes should be expected between minor versions.
+
+## Troubleshooting
+
+### Enabling Debug logging
+
+This package integrates with Python's [logging facility](https://docs.python.org/3/library/logging.html) through a Rust crate called [`pyo3_log`](https://docs.rs/pyo3-log/latest/pyo3_log/). The quickest way to get started is to just enable debug logging:
+
+```python
+import logging
+logging.basicConfig(level=logging.DEBUG)
+```
+
+Because this is implemented with Rust, there are some important differences in regards to log levels and filtering.
+
+#### The `TRACE` log level
+
+Rust has a `TRACE` log level that doesn't exist in Python. It is less severe than `DEBUG` and is set to a value of 5. While the `DEBUG` level is recommended for troubleshooting, you can choose to target `TRACE` level logs and below like so:
+
+```python
+import logging
+logging.basicConfig(level=5)
+```
+
+#### Runtime Configuration and Caching
+ 
+`pyo3_log` caches loggers and their level filters to improve performance. This means that logger re-configuration done at runtime may cause logs to log or not log as expected in certain situations. If this is a concern, [this section of the pyo3_log documentation](https://docs.rs/pyo3-log/latest/pyo3_log/#performance-filtering-and-caching) goes into more detail.
```

Comment:
```suggestion
`pyo3_log` caches loggers and their level filters to improve performance. This means that logger re-configuration done at runtime may cause unexpected logging behavior in certain situations. If this is a concern, [this section of the pyo3_log documentation](https://docs.rs/pyo3-log/latest/pyo3_log/#performance-filtering-and-caching) goes into more detail.
```

Should we expose a simple binding to `reset_logging` and indicate that here? Logging is simple when it works but can be maddening when it doesn't, and an ability for the user to dig themselves out of a caching problem will likely be appreciated. If that sounds good I can add the commit.